packages/villus/src/utils/common.ts

```diff
@@ -16,13 +17,17 @@ export function toWatchableSource<T = any>(value: Ref<T> | Record<string, any>):
 
 // Uses same component provide as its own injections
 // Due to changes in https://github.com/vuejs/vue-next/pull/2424
+// todo: maybe better to modify func name and add manual client args here
 export function injectWithSelf<T>(symbol: InjectionKey<T>, onMissing: () => Error): T {
```

Comment:
Let's change this like you suggested. Villus only uses this to inject the client so it is safe to completely re-write to just resolve the client either from the active one or the injection.

```suggestion
export function resolveClient(): Client {
```